Fix hover effect on Log in and Sign up nav items

shanrauf opened the pull request:
I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] is from a uniquely-named feature branch and has been rebased on top of the latest master. (If I was asked to make more changes, I have made sure to rebase onto master then too)
* [x] is descriptively named and links to an issue number: `Applies fixes from #908 to "Log in" and "Sign up"`

catarak commented:
thanks for working on this. the hover color is fixed, however, it appears the spacing is a little off:
<img width="355" alt="Screen Shot 2019-05-28 at 6 06 01 PM" src="https://user-images.githubusercontent.com/6063380/58515606-63607a00-8173-11e9-880a-1eb7c2145948.png">
there's more space between "log in" and "or" than "sign up" and "or"

shanrauf commented:
My apologies for putting this off for a whole week... I fixed the unequal spacing between Log in and Sign up. Please let me know if there are any other issues with this commit.

shanrauf commented:
Apologies for the weird commit history, had trouble with Git... The unequal spacing was caused by a margin that was originally placed on all nav items seemingly to make the dropdowns look nicer. However, since Log in and Sign up don't have dropdowns, I just deleted the margin placed on all nav items, and added a margin to the triangles (which will only affect the necessary nav items). 

Please let me know if there are any other issues.

catarak commented on client/components/Nav.jsx:
```diff
@@ -531,11 +531,19 @@ class Nav extends React.PureComponent {
-                <span className="nav__item-spacer">or</span>
-                <Link to="/signup">Sign up</Link>
-              </p>
+              <Link to="/login">
```
hey, so this html is not semantically correct—rather than inserting a `<button>` within the link, the styles for these should be updated.

shanrauf replied:
Got it, I'll keep this in mind and make changes.

shanrauf commented:
My new commit doesn't weirdly add a button in the <Link> element anymore. The hover effect I added to all nav__item-headers is a bit redundant because right above it there is something similar for the left nav, but I don't see any other option. I als oremoved the nav__item class on the right nav <li> elements, because f you keep that class, then there is the same problem from #908 where the hover effect is applied but the link is not even clickable. Please let me know if you want anything changed!

catarak commented:
there's something funky happening with the git history on this branch...

shanrauf commented:
Apologies once again... Everything should be fixed, so please let me know if anything else is wrong. I appreciate you dealing with this inefficiency, I'm learning a lot...

catarak commented:
@shanrauf no worries!

catarak commented:
this is working great for me. thanks for working on this!
